src/c++/library/grpc_client.cc

```diff
@@ -91,7 +91,7 @@ GetStub(
       std::stoul(GetEnvironmentVariableOrDefault(
           "TRITON_CLIENT_GRPC_CHANNEL_MAX_SHARE_COUNT", "6"));
   const auto& channel_itr = grpc_channel_stub_map_.find(url);
-  if (channel_itr != grpc_channel_stub_map_.end()) {
+  if ((channel_itr != grpc_channel_stub_map_.end()) && (!dont_use_cached_channel)){
```

Comment:
Can you call the variable something like `reuse_connection`? Double negation is a bit twisted.